props/prop_base.h

```diff
@@ -379,37 +373,23 @@ class PropBase : CommandParser, Looper, protected SaberBase {
     SaveVolumeIfNeeded();
   }
 
-  // Select preset (font/style)
-  virtual void SetPreset(int preset_num, bool announce) {
-    TRACE(PROP, "start");
+  bool BladeOff() {
 #ifdef IDLE_OFF_TIME
     last_on_time_ = millis();
 #endif
     bool on = SaberBase::IsOn();
     if (on) Off();
-    SaveColorChangeIfNeeded();
-    // First free all styles, then allocate new ones to avoid memory
-    // fragmentation.
+    return on;
+  }
+
+  void UnsetCurrentStyles(int preset_num) {
 #define UNSET_BLADE_STYLE(N) \
     delete current_config->blade##N->UnSetStyle();
     ONCEPERBLADE(UNSET_BLADE_STYLE)
     current_preset_.SetPreset(preset_num);
-    if (announce) {
-      if (current_preset_.name.get()) {
-        SaberBase::DoMessage(current_preset_.name.get());
-      } else {
-        char message[64];
-        strcpy(message, "Preset: ");
-        itoa(current_preset_.preset_num + 1,
-             message + strlen(message), 10);
-        strcat(message, "\n");
-        strncat(message + strlen(message),
-                current_preset_.font.get(), sizeof(message) - strlen(message));
-        message[sizeof(message) - 1] = 0;
-        SaberBase::DoMessage(message);
-      }
-    }
+  }
 
+  void SetBladeLength() {
```

Comment:
Hmm, this probably don't belong in SetPreset() at all.
We should probably do this separately.

Reply:
Actually never mind.
I looked at this functoin, and it doesn't actually *do* anything except for restoring the savestate.
Everything else is done in AllocatedBladeStyles(). Breaking this out from AllocateBladeStyles doesn't really work, so just put together with AllocateBladeStyles().